packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2464,6 +2489,397 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+  ///
+  /// * [maybeOf], which is similar, but returns null if no
+  ///   [MenuAcceleratorCallbackBinding] is found.
+  static MenuAcceleratorCallbackBinding of(BuildContext context) {
+    final MenuAcceleratorCallbackBinding? result = maybeOf(context);
+    assert(() {
+      if (result == null) {
+        throw FlutterError(
+          'MenuAcceleratorWrapper.of() was called with a context that does not '
+          'contain a MenuAcceleratorWrapper in the given context.\n'
+          'No MenuAcceleratorWrapper ancestor could be found in the context that '
+          'was passed to MenuAcceleratorWrapper.of(). This can happen because '
+          'you are using a widget that looks for a MenuAcceleratorWrapper '
+          'ancestor, and do not have a MenuAcceleratorWrapper widget ancestor.\n'
+          'The context used was:\n'
+          '  $context',
+        );
+      }
+      return true;
+    }());
+    return result!;
+  }
+}
+
+/// The type of builder function used for building a [MenuAcceleratorLabel]'s
+/// [MenuAcceleratorLabel.builder] function.
+///
+/// {@template flutter.material.menu_anchor.menu_accelerator_child_builder.args}
+/// The arguments to the function are as follows:
+///
+/// * The `context` supplies the [BuildContext] to use.
+/// * The `label` is the [MenuAcceleratorLabel.label] attribute for the relevant
+///   [MenuAcceleratorLabel] with the accelerator markers stripped out of it.
+/// * The `index` is the index of the accelerator character within the `label`
+///   that applies to this accelerator. If it is -1, then the accelerator should
+///   not be highlighted. Otherwise, the given character should be highlighted
+///   somehow in the rendered label (typically with an underscore).
+/// {@endtemplate}
+///
+/// See also:
+///
+/// * [MenuAcceleratorLabel.defaultLabelBuilder], which is the implementation
+///   used as the default value for [MenuAcceleratorLabel.builder].
+typedef MenuAcceleratorChildBuilder = Widget Function(
+  BuildContext context,
+  String label,
+  int index,
+);
+
+/// A widget that draws the label text for a menu item (typically a
+/// [MenuItemButton] or [SubmenuButton]) and renders its child with information
+/// about the currently active keyboard accelerator.
+///
+/// On platforms other than macOS and iOS, this widget listens for the Alt key
+/// to be pressed, and when it is down, will update the label by calling the
+/// builder again with the position of the accelerator in the label string.
+/// While the Alt key is pressed, it registers a shortcut with the
+/// [ShortcutRegistry] mapped to a [VoidCallbackIntent] containing the callback
+/// defined by the nearest [MenuAcceleratorCallbackBinding].
+///
+/// Because the accelerators are registered with the [ShortcutRegistry], any
+/// other shortcuts in the widget tree between the [primaryFocus] and the
+/// [ShortcutRegistry] that define Alt-based shortcuts using the same keys will
+/// take precedence over the accelerators.
```

Comment:
I was wondering about this while reading these docs, good idea explaining it here.